packages/cardpay-sdk/sdk/safes.ts

```diff
@@ -217,6 +226,63 @@ export async function viewSafe(network: 'gnosis' | 'xdai' | 'sokol', safeAddress
 export default class Safes implements ISafes {
   constructor(private layer2Web3: Web3, private layer2Signer?: Signer) {}
 
+  async createSafe(
+    owners: string[],
+    threshold: number,
+    tokenAddress: string,
+    saltNonce?: string,
+    txnOptions?: TransactionOptions
+  ) {
```

Comment:
generally all the SDK functions have the ability to accept a txn hash as well so that if the txn is still not confirmed the SDK function will return a promise after the txn has been mined. This allows us to resume a workflow that is in progress and waiting for txns to be mined. Here are a bunch of examples: https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/prepaid-card/base.ts#L108-L111, https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/prepaid-card/base.ts#L183-L186, 
https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/revenue-pool/base.ts#L114-L117,
https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/revenue-pool/base.ts#L207-L214,
https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/prepaid-card-market-v-2/base.ts#L59-L62,
https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/prepaid-card-market-v-2/base.ts#L308-L311